### ISSUE TITLE
clippy subcommand incorrectly warns about unknown lints in subcrates

### ISSUE THREAD

**Issue by debris** [Manishearth/rust-clippy#1331]
steps to reproduce:

1. Create new project and enter it
    ```
    cargo new foo
    cd foo
    ```

2. Create subproject and add it to `Cargo.toml`
    ```
    cargo new bar
    ```

    ```
    [dependencies]
    foo = { path = "foo" }
    ```

3. Add some code to `foo/src/lib.rs`
    ```rust
    #[cfg_attr(feature="cargo-clippy", allow(doc_markdown))]
    /// some markdown which require allow(doc_markdown) here
    pub fn foo() {}
    ```

4. Run clippy for `foo`
     ```
     cargo clippy -p foo
     ```

     Everything works as expected

5. Run clippy for `bar`
     ```
     cargo clippy -p bar
     ```

    Surprisingly clippy returns warning:

    ```
    warning: unknown lint: `doc_markdown`, #[warn(unknown_lints)] on by default
       --> /path_to_bar/foo/src/lib.rs:0:3
        |
    289 | #[cfg_attr(feature="cargo-clippy", allow(doc_markdown))]
        |                                          ^^^^^^^^^^^^^^^
    ```
